mage_ai/ai/llm_pipeline_wizard.py

```diff
@@ -216,15 +323,54 @@ async def async_generate_block_with_description(
             function_args = json.loads(response_message["function_call"]["arguments"])
             block_type, block_language, pipeline_type, config = self.__load_template_params(
                 function_args)
+            if 'data_source' not in config.keys() and \
+                    ('action_type' not in config.keys() or 'axis' not in config.keys()):
+                # Fill customized code for default template.
+                # Logic to check default template should be same
+                # as __fetch_transformer_templates function,
+                customized_logic = await \
+                    self.__async_llm_generate_customized_code_with_base_template(
+                        block_description,
+                        block_language,
+                        PROMPT_FOR_CUSTOMIZED_CODE_WITH_BASE_TEMPLATE
+                    )
+                if 'code' in customized_logic.keys():
+                    config["existing_code"] = customized_logic.get("code")
+                block_code = fetch_template_source(
+                        block_type=block_type,
+                        config=config,
+                        language=block_language,
+                        pipeline_type=pipeline_type,
+                    )
+            else:
+                try:
+                    block_code_template = fetch_template_source(
+                            block_type=block_type,
+                            config=config,
+                            language=block_language,
+                            pipeline_type=pipeline_type,
+                        )
+                    block_code = await self.async_create_customized_code_in_block(
+                        block_code_template,
+                        block_language,
+                        block_description)
+                except TemplateNotFound:
+                    # Use default template if template not found and
+                    # ask LLM to fully generate the customized code.
+                    customized_logic = await \
+                        self.__async_llm_generate_customized_code_with_base_template(
+                            block_description,
+                            block_language,
+                            PROMPT_FOR_CUSTOMIZED_CODE_WITH_BASE_TEMPLATE
+                        )
+                    block_code = template_env.get_template('transformers/default.jinja').render(
```

Comment:
Let's put the template logic inside `template.py`. The relative path of loading template is not correct.

Reply:
Good point! I moved this logic into a new function called `fetch_transformer_default_template` in `template.py`. Also create a new function named `is_default_transformer_template` in `template.py` and used it at line 349